### PR TITLE
feat(check): introduce page.check/uncheck

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/dom.ts [hunk 1]
```diff
@@ -459,6 +459,46 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
...
+    await this._setChecked(true);
+  }
+
+  async uncheck() {
+    await this._setChecked(false);
+  }
+
+  private async _setChecked(state: boolean) {
+    const isCheckboxChecked = async (): Promise<boolean> => {
+      return this.evaluate((node: Node) => {
```
Comment (dgozman) [inline src/dom.ts]: Use `this._evaluateInUtility`.

File: test/click.spec.js [hunk 2]
```diff
@@ -405,4 +405,41 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       expect(await page.evaluate('document.querySelector("#button").style.left')).toBe('0px');
     });
   });
+
+  describe('Page.check', function() {
```
Comment (dgozman) [inline test/click.spec.js]: We should follow up with checking radio buttons (IIRC, WebDriver does it).

File: test/click.spec.js [hunk 3]
```diff
@@ -405,4 +405,41 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
...
+      await page.uncheck('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(false);
+    });
+    it('should not uncheck the unchecked box', async({page}) => {
+      await page.setContent(`<input id='checkbox' type='checkbox'></input>`);
+      await page.uncheck('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(false);
+    });
+    it('should check the box by label', async({page}) => {
+      await page.setContent(`<label for='checkbox'><input id='checkbox' type='checkbox'></input></label>`);
```
Comment (dgozman) [inline test/click.spec.js]: Could you please add the following tests?

Checkbox not inside the label:
`<label for='checkbox'>Text</label><div><input id='checkbox' type='checkbox'></input></div>`

Checkbox inside a label without for=.
`<label>Text<span><input id='checkbox' type='checkbox'></input></span></label>`

**Review by dgozman — APPROVED** [review]

File: docs/api.md [hunk 4]
```diff
@@ -729,6 +731,18 @@ To disable authentication, pass `null`.
 
 Get the browser context that the page belongs to.
 
+#### page.check(selector, [options])
+- `selector` <[string]> A selector to search for checkbox to check. If there are multiple elements satisfying the selector, the first will be checked.
+- `options` <[Object]>
+  - `waitFor` <"visible"|"hidden"|"any"|"nowait"> Wait for element to become visible (`visible`), hidden (`hidden`), present in dom (`any`) or do not wait at all (`nowait`). Defaults to `visible`.
+  - `timeout` <[number]> Maximum navigation time in milliseconds, defaults to 30 seconds, pass `0` to disable timeout. The default value can be changed by using the [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) method.
```
Comment (aslushnikov) [inline docs/api.md]: remove "navigation"

Comment (pavelfeldman) [reply docs/api.md]: Done

File: src/dom.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/dom.ts]: Done

File: test/click.spec.js (hunk 3 above)
Comment (pavelfeldman) [reply test/click.spec.js]: Done

Comment (pavelfeldman) [reply test/click.spec.js]: Done

File: test/click.spec.js (hunk 2 above)
Comment (pavelfeldman) [reply test/click.spec.js]: Added radio into the mix

**Review by dgozman — APPROVED** [review]
Wonderful!

**Pull request merged**